src/behaviours/quitting-and-restarting-the-app/opening-application-window-using-tray.test.ts

```diff
@@ -0,0 +1,163 @@
+} from "../../main/start-main-application/lens-window/application-window/create-lens-window.injectable";
+
+import { flushPromises } from "../../common/test-utils/flush-promises";
+import type { DiContainer } from "@ogre-tools/injectable";
+
+describe("opening application window using tray", () => {
+  describe("given application has started", () => {
+    let applicationBuilder: ApplicationBuilder;
+
+    let createElectronWindowMock: AsyncFnMock<
+      (configuration: LensWindowConfiguration) => ElectronWindow
+    >;
+
+    let expectWindowsToBeOpen: (windowIds: string[]) => void;
+    let resolveOpeningOfWindow: (windowId: string) => Promise<void>;
+
+    beforeEach(async () => {
+      applicationBuilder = getApplicationBuilder().beforeApplicationStart(
+        ({ mainDi }) => {
+          createElectronWindowMock = asyncFn();
+
+          mainDi.override(
+            createElectronWindowForInjectable,
+
+            () => (configuration) => () =>
+              createElectronWindowMock(configuration),
+          );
+
+          expectWindowsToBeOpen = expectWindowsToBeOpenFor(mainDi);
+
+          resolveOpeningOfWindow = resolveOpeningOfWindowFor(
+            createElectronWindowMock,
+          );
+        },
+      );
+
+      const renderPromise = applicationBuilder.render();
+
+      await flushPromises();
+
+      await resolveOpeningOfWindow("splash");
+      await resolveOpeningOfWindow("only-application-window");
+
+      await renderPromise;
+    });
+
+    it("only an application window is open", () => {
+      expectWindowsToBeOpen(["only-application-window"]);
+    });
+
+    describe("when the application window is closed", () => {
+      beforeEach(() => {
+        const applicationWindow = applicationBuilder.dis.mainDi.inject(
+          applicationWindowInjectable,
+        );
+
+        applicationWindow.close();
+      });
+
+      it("no windows are open", () => {
+        expectWindowsToBeOpen([]);
+      });
+
+      describe("when an application window is reopened using tray", () => {
+        beforeEach(() => {
+          applicationBuilder.tray.click("open-app");
+        });
+
+        it("still no windows are open", () => {
+          expectWindowsToBeOpen([]);
+        });
+
+        describe("given opening of splash window has not finished yet, but another attempt to open the application is made", () => {
+          beforeEach(() => {
+            createElectronWindowMock.mockClear();
+
+            applicationBuilder.tray.click("open-app");
+          });
+
+          it("does not open any new windows", () => {
+            expect(createElectronWindowMock).not.toHaveBeenCalled();
+          });
+        });
+
+        describe("when opening of splash window resolves", () => {
+          beforeEach(async () => {
+            await resolveOpeningOfWindow("splash");
+          });
+
+          it("still only splash window is open", () => {
+            expectWindowsToBeOpen(["splash"]);
+          });
+
```

Comment:
I would like to see another test along the lines of "given another attempt to open the application is made > shows window immediately without waiting for it to load".

I believe that is what the previous behaviour was and is useful when you get a crash before the window has reported that it is loaded, that way we can open the dev tools and look at the logs.